src/refmvs.rs

```diff
@@ -85,15 +85,28 @@ pub struct refmvs_mvpair {
     pub mv: [mv; 2],
 }
 
+/// For why this unaligned, see the aligned [`refmvs_block`] below.
 #[derive(Copy, Clone)]
 #[repr(C, packed)]
-pub struct refmvs_block {
+pub struct refmvs_block_unaligned {
     pub mv: refmvs_mvpair,
     pub r#ref: refmvs_refpair,
     pub bs: uint8_t,
     pub mf: uint8_t,
 }
 
+/// In C, `struct refmvs_block` is both aligned and packed,
+/// but this (aligned types within a packed type) is not yet allowed in Rust
+/// (see [rust-lang/rust#59154](https://github.com/rust-lang/rust/issues/59154)),
+/// as different C compilers treat this differently.
+/// To get around this limitation, we split `struct refmvs_block`
+/// into an inner packed [`refmvs_block_unaligned`]
+/// and an outer aligned [`refmvs_block`]
+/// that is just a wrapper over the real [`refmvs_block_unaligned`].
+#[derive(Copy, Clone)]
+#[repr(C, align(4))]
+pub struct refmvs_block(pub refmvs_block_unaligned);
```

Comment:
Does the inner value need `pub`?

Reply:
It needs to at least be `pub(crate)`, but like https://github.com/memorysafety/rav1d/pull/233#discussion_r1187959647, we haven't been cleaning up `pub`s yet, and probably won't be for a little while.